walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,81 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private void addTimeRange(List<TimeRange> timeList, int startTime, int endTime, boolean includeEnd) {
+        timeList.add((TimeRange.fromStartEnd(  
+            startTime,
+            endTime,
+            includeEnd)));
+    }
+
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> eventTimeRanges = new ArrayList<>();
+        List<TimeRange> openTimeRanges = new ArrayList<>();
+
+        if(request.getAttendees() != null && !request.getAttendees().isEmpty()) {
+            if(TimeRange.WHOLE_DAY.duration() >= request.getDuration()) {
```

Comment:
nit: for readability, it seems like the order of this condition should be reversed. You care how the duration compares to WHOLE_DAY, so I would structure it in that order.